lib/spack/spack/binary_distribution.py

```diff
@@ -159,33 +500,27 @@ def __init__(self, cache_root: Optional[str] = None):
         self._index_contents_key = "contents.json"
 
         # a FileCache instance storing copies of remote binary cache indices
-        self._index_file_cache: Optional[file_cache.FileCache] = None
+        self._index_file_cache: file_cache.FileCache = file_cache.FileCache(self._index_cache_root)
```

Comment:
Why not leave this Optional?

Reply:
I switched it to appease the type hinter, which was probably complaining for a good reason.  It seems to me we have to have one of these, of what use is leaving it `Optional`?

Reply:
I think the main advantage is we don't need an extra flag to check if the cache is initialized. I am not sure why the type checker would have complained here..it is annotated and follows the annotation everywhere.

Reply:
I think it complains because it could be `None` and we don't check it before every use.  It's seems not smart enough to see that at some point we called the method that initializes it.  Or maybe there was a path that could leave it uninitialized, but it didn't look like that was the case.